test/embedding.jl

```diff
@@ -0,0 +1,12 @@
+# test that the embedding example runs without error
+let
+    if is_windows()
+        embedding_bin = joinpath(JULIA_HOME,"embedding.exe")
+    else
+        embedding_bin = joinpath(JULIA_HOME,"..","libexec","embedding")
```

Comment:
People objected (understandably) to globally installing a binary called "embedding":

https://github.com/JuliaLang/julia/pull/18688#discussion_r80617896

So I picked libexec, because it is designed for internal binaries (per the FHS, at least).